frontend/src/app/supportAdmin/UnarchivePatients/UnarchivePatient.tsx

```diff
@@ -0,0 +1,258 @@
+import React, { useEffect, useState } from "react";
+import { useNavigate, useParams } from "react-router-dom";
+
+import {
+  ArchivedStatus,
+  useGetOrganizationsQuery,
+  useGetOrganizationWithFacilitiesLazyQuery,
+  useGetPatientsByFacilityWithOrgLazyQuery,
+  useGetPatientsCountByFacilityWithOrgLazyQuery,
+} from "../../../generated/graphql";
+import { LoadingCard } from "../../commonComponents/LoadingCard/LoadingCard";
+import { showError } from "../../utils/srToast";
+import { Option } from "../../commonComponents/Select";
+import { useSelectedFacility } from "../../facilitySelect/useSelectedFacility";
+import { useDocumentTitle } from "../../utils/hooks";
+import { unarchivePatientTitle } from "../pageTitles";
+
+import UnarchivePatientInformation from "./UnarchivePatientInformation";
+import UnarchivePatientFilters from "./UnarchivePatientFilters";
+
+export type UnarchivePatientOrganization = {
+  internalId: string;
+  externalId: string;
+  name: string;
+  facilities?: {
+    id: string;
+    name: string;
+  }[];
+};
+
+export type UnarchivePatientFacility = {
+  id: string;
+  name: string;
+};
+
+export type UnarchivePatientPatient = {
+  internalId: string;
+  firstName: string;
+  lastName: string;
+  middleName: string;
+  birthDate: string;
+  isDeleted: boolean;
+  facility: UnarchivePatientFacility | null;
+};
+
+export interface UnarchivePatientState {
+  pageUrl: string;
+  entriesPerPage: number;
+  orgId: string;
+  facilityId: string;
+  patientsCount: number | undefined;
+  patients: UnarchivePatientPatient[] | undefined;
+  facilities: UnarchivePatientFacility[];
+}
+
+export const initialState: UnarchivePatientState = {
+  pageUrl: "/admin/unarchive-patient",
+  entriesPerPage: 20,
+  orgId: "",
+  facilityId: "",
+  patientsCount: undefined,
+  patients: undefined,
+  facilities: [],
+};
+
+const getOrgExternalIdWithInternalId = (
+  orgs: UnarchivePatientOrganization[],
+  internalId: string
+) => {
+  let org = orgs.find((org) => org.internalId === internalId);
+  return org?.externalId;
+};
+
+const UnarchivePatient = () => {
+  const navigate = useNavigate();
+  const [localState, updateLocalState] =
+    useState<UnarchivePatientState>(initialState);
+  const { pageNumber } = useParams();
+  const currentPage = pageNumber ? +pageNumber : 1;
+  const [activeFacility] = useSelectedFacility();
+
+  useDocumentTitle(unarchivePatientTitle);
+
+  const {
+    data: orgsResponse,
+    loading: loadingOrgs,
+    error,
+  } = useGetOrganizationsQuery({
+    fetchPolicy: "no-cache",
```

Comment:
If the odds of deleting an organization is low this would be a good candidate to use a cache policy. Let me know what you think!

Reply:
ooo ok! I'll remove the fetchpolicy since it looks like `cache-first` is the default when not specifying: https://www.apollographql.com/docs/react/data/queries/#setting-a-fetch-policy